backend/entities/views.py

```diff
@@ -1,250 +1,250 @@
+            {"message": "Organization deleted successfully"}, status.HTTP_200_OK
+        )
+
+
+class OrganizationApplicationViewSet(viewsets.ModelViewSet[OrganizationApplication]):
+    queryset = OrganizationApplication.objects.all()
+    serializer_class = OrganizationApplicationSerializer
+    pagination_class = CustomPagination
+
+
+class OrganizationEventViewSet(viewsets.ModelViewSet[OrganizationEvent]):
+    queryset = OrganizationEvent.objects.all()
+    serializer_class = OrganizationEventSerializer
+    pagination_class = CustomPagination
+
+
+class OrganizationMemberViewSet(viewsets.ModelViewSet[OrganizationMember]):
+    queryset = OrganizationMember.objects.all()
+    serializer_class = OrganizationMemberSerializer
+    pagination_class = CustomPagination
+
+
+class OrganizationResourceViewSet(viewsets.ModelViewSet[OrganizationResource]):
+    queryset = OrganizationResource.objects.all()
+    serializer_class = OrganizationResourceSerializer
+    pagination_class = CustomPagination
+
+
+class GroupViewSet(viewsets.ModelViewSet[Group]):
+    queryset = Group.objects.all()
+    serializer_class = GroupSerializer
+    pagination_class = CustomPagination
+
+    def list (self, request: Request, *args: str, **kwargs: int) -> Response:
+        serializer = self.get_serializer(self.queryset, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    
+    def create(self, request: Request, *args: str, **kwargs: int) -> Response:
+        serializer = self.get_serializer(data=request.data)
+        if serializer.is_valid():
+            group = serializer.save(created_by=request.user)
+            data = {"message": f"New Group created: {serializer.data}"}
+            return Response(data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+    
+    def retrieve(self, request: Request, *args: str, **kwargs: int) -> Response:
+        group = self.queryset.filter(id=kwargs["pk"])
+        if group:
+            serializer = self.get_serializer(group)
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        return Response({"error": f"Group not found"}, status=status.HTTP_404_NOT_FOUND)
+    
+    def partial_update(self, request: Request, *args: str, **kwargs: int) -> Response:
+        group = self.queryset.filter(id=kwargs["pk"])
+        if group is not None:
+            return Response({"error: Group not found"}, status.HTTP_404_NOT_FOUND)
+        
+        if request.user != group.created_by:
+            return Response({"error": "You are not authorized to update this group"}, status.HTTP_401_UNAUTHORIZED)
+        
+        serializer = self.get_serializer(group, data=request.data, partial=True)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+    
+    def destroy(self, request: Request, *args: str, **kwargs: int) -> Response:

```

Comment:
@thiagogquinto 
When we are deleting the group, we want to make sure that only the creator is allowed to do that. 

@andrewtavis 
At the moment, when we delete the group, we set the deletion_date. But there should be more to this, like:

1. Delete the group and remove the `deletion_date` field
2. Set fields to blank/NULL on deletion

Reply:
I don't think we need to worry as much about keeping minor data for groups as we do for for orgs. I think the first option would be fine? Saves on storage as well.

Reply:
But would you suggest either, @to-sta?

Reply:
I guess it's not necessary to set values to blank/NULL. I would suggest to delete the entry and remove the deletion_date field.